### DISCUSSION TITLE
Official way to delete a backoffice user

### DISCUSSION THREAD

**Discussion by mcl-sz** [umbraco/Umbraco-CMS#11239] [Features and ideas]
In the backoffice there is no way to delete a user once a user has loggedin. While it is possible to disable account, it is not sufficient in our case to comply with the GPDR. In this article https://piotrbach.com/how-to-delete-umbraco-cms-user-permanently Piotr Bach explains how to delete an user from the database but it would be nice if there is an "official" way to delete backoffice users. Preferably through a button in the backoffice and userservice.

**Comment by bjarnef** [comment]
Eventually the delete action could be a button on the user page (and maybe also as bulk action?).
Maybe the delete action should only be available when a user already is disabled making it a bit harder accidentally deleting a user.

**Comment by aidinfanni** [comment]
This is absolutely required or Umbraco isn't really GDPR compliant.

**Reply by skttl** [reply]
Rename the user to John Doe, set the email to johndoe@johndoe.com and then disable the user. Then you are GDPR compliant :)

**Reply by aidinfanni** [reply]
Yes, but when you have JohnDoe113, it becomes a problem.

**Comment by mcl-sz** [comment]
A CMS where it is not even possible to delete a user has something shabby...

**Comment by LennardF1989** [comment]
Just a random thought: the problem with deleting users is because it affects everything they touched in the backoffice from an Audit trail perspective as well as (on a technical level) the foreign keys in the database.

So instead of deleting, remap all that the user "owns" to another user id. You can then technically "merge" users together, effectively removing the old user and maintaining the integrity (in a broad sense) of the data.

You could just add a user to your Umbraco instance that has no other purpose than being a piggy back for deleted users. Or maybe Umbraco can come with that one out of the box, unable to login by any means, and hide it if you don't want to see it.
